main/src/com/google/refine/grel/controls/ForRange.java

```diff
@@ -83,32 +83,46 @@ public Object call(Properties bindings, Evaluable[] args) {
                 long step = ((Number) stepO).longValue();
                 double to = ((Number) toO).doubleValue();
 
-                if (!((from < to && step > 0) || (from > to && step < 0))) {
+                if (step == 0) {
                     return new EvalError(EvalErrorMessage.invalid_arg());
                 }
 
-                int rangeSize = (int) Math.ceil(Math.abs(from - to) / Math.abs(step));
-                for (int i = 0; i < rangeSize; i++) {
-                    bindings.put(indexName, from);
-                    Object r = args[4].evaluate(bindings);
-                    results.add(r);
-                    from += step;
+                if (step <= Math.abs(from - to)) {
```

Comment:
This test is unnecessary. It will be handled naturally by the loop logic (ie if step > (from-to) there will be zero iterations.

Reply:
I misunderstood the meaning of zero iterations. I have adjusted the code.